### ISSUE TITLE
Show workspace parameters in Workspaces table view

### ISSUE THREAD

**Issue by michaelbrewer** [coder/coder#12447]
# Problem statement

As a developer when i have multiple workspaces open for different projects, i don't know what is the the checked out project. 

<img width="1395" alt="Screenshot 2024-03-06 at 3 44 06 PM" src="https://github.com/coder/coder/assets/5442469/a79959a4-29b6-443d-a902-c6a76d89a96e">

Ideally you should be able to see or filter by Workspace Parameters

<img width="1143" alt="Screenshot 2024-03-06 at 3 45 10 PM" src="https://github.com/coder/coder/assets/5442469/2a715c33-5e78-4a8c-be9a-13870419bf7a">

**Comment by kylecarbs** [comment]
Kinda interesting that a combination of parameters would replace a name.

Good idea!

**Comment by michaelbrewer** [comment]
Yes, currently these workspaces are opened from "Open in Coder" or via an inhouse Chrome extension.

As a chrome extension

<img width="425" alt="Screenshot 2024-03-06 at 3 48 16 PM" src="https://github.com/coder/coder/assets/5442469/80398702-bbee-4bdd-8caa-355da2d839d7">

As from Open in Coder

<img width="577" alt="Screenshot 2024-03-06 at 3 48 36 PM" src="https://github.com/coder/coder/assets/5442469/d2c6e4fb-55b4-46b0-a7df-7f7c8b14ce1d">

**Comment by BrunoQuaresma** [comment]
@michaelbrewer thanks for bringing this idea to us 🙏. I see how this can be useful.

I see this can be challenging since a workspace can have many parameters, and their labels and values can be quite long. Would you have any other product reference that would display similar data to the users or would you have ideas on how we can do that? In the meantime, I'm going to think about this and see if I can come up with something 🤔.

**Comment by michaelbrewer** [comment]
For now : 
I can pass in the "repo_name" to refill the workspace name for now. 
Also for an open workspace I will see if I can add meta-data fields to identify what's open quickly. 

Feature request wise:

Generating a name from the parameters could be too long, maybe just allow for a prefix to see the name might help. Ie: (repo_name)-(generated-name)

Allowing for filtering / search on workspace parameters would help admins and allowing users to select a parameter to the list view would be great like the git_url parameter. 

I can see that this could extend to seeing the list of workspaces in vscode and JetBrains extensions.

**Comment by matifali** [comment]
Related to #11837

**Comment by BrunoQuaresma** [comment]
@michaelbrewer do you think this would be helpful to you https://github.com/coder/coder/pull/12694?

**Comment by michaelbrewer** [comment]
Could be i will try it out. I assume you can use this to filter workspaces by a parameter name? like repo url.

**Comment by michaelbrewer** [comment]
@BrunoQuaresma  - would this filter pick up workspace metadata as well?  For example we could flag that a workspace has uncommitted work, so we don't want to delete these unnecessarily.

https://github.com/coder/coder/discussions/9477

**Comment by BrunoQuaresma** [comment]
cc.: @Emyrk

**Comment by Emyrk** [comment]
@michaelbrewer no. It only searches parameters defined by the template.

**Comment by michaelbrewer** [comment]
Nice @BrunoQuaresma  i tested it how, only problem is that requires exact match, so for git_repo urls you have to do

`owner:me param:"git_repo=https://github.com/microsoft/vscode-remote-try-go"`

But it would be nice to do a contain search:

`owner:me param:"git_repo=*vscode-remote-try-go"`

**Comment by Emyrk** [comment]
@michaelbrewer I can reopen this?
https://github.com/coder/coder/pull/12695

**Comment by michaelbrewer** [comment]
@Emyrk possibly. If there is no concerns with performance. 

Visually having an option to select additional fields would be great. 

Search and filtering is useful for looking across an entire team, group or organization

**Comment by Emyrk** [comment]
@michaelbrewer I closed this at the time as it was not needed for the original purpose.

I can reopen this if it solves your use case

**Comment by Emyrk** [comment]
Got it. If updating the list view to see parameters is the better approach, I would rather not expand the filtering at this time.

I want to avoid adding it to the filter as it adds inconsistency in the other filter params. As it would be the only filter term to support `*`. Which is confusing.

**Comment by tillig** [comment]
Search by repo name implies there's a repo provided during checkout and enforces a one-repo-per-workspace development model that doesn't work so well if you're trying to use the same environment to work on multiple related repos. That's not a huge issue, but it's something to be aware of when looking at filtering options including "by repo."

**Comment by michaelbrewer** [comment]
for collections of repos, users creating a workspace should name it clearly. But short lived workspaces or tasks that can operate on a single repo having access to fields to filter on helps, as a workspace could have a number of custom parameters. like a custom parameter for a collection of repos to be checked out all at once for a specific project

**Comment by michaelbrewer** [comment]
Still no way show additional field. Its a common behavior to use a generated name for a workspace, and then you have no clue what the workspace is for.
